context/llpcPipelineContext.cpp

```diff
@@ -416,11 +416,24 @@ void PipelineContext::SetOptionsInPipeline(
             shaderOptions.waveBreakSize = pShaderInfo->options.waveBreakSize;
 #endif
 
-            bool loadScalarizerEnabled = EnableScalarLoad;
+            shaderOptions.loadScalarizerThreshold = 0;
+            if (EnableScalarLoad)
+            {
+                shaderOptions.loadScalarizerThreshold = ScalarThreshold;
+            }
 #if LLPC_CLIENT_INTERFACE_MAJOR_VERSION >= 33
-            loadScalarizerEnabled |= pShaderInfo->options.enableLoadScalarizer;
+            if (pShaderInfo->options.enableLoadScalarizer)
+            {
+                if (pShaderInfo->options.scalarThreshold!=0)
+                {
+                    shaderOptions.loadScalarizerThreshold = pShaderInfo->options.scalarThreshold;
+                }
+                else
+                {
+                    shaderOptions.loadScalarizerThreshold = 0xFFFFFFFF;
```

Comment:
static cl::opt<unsigned> ScalarThreshold("scalar-threshold",
                                         cl::desc("The threshold for load scalarizer"),
                                         cl::init(0xFFFFFFFF));

I think you can use InvalidValue. If it couldn't express the meaning of "max". You can define: static const uint32_t MaxScalarThreshold = 0xFFFFFFFF.